scripts/gen_lib.lua

```diff
@@ -0,0 +1,71 @@
+lpeg = require('lpeg')
```

Comment:
> I'm talking about sensible defaults; how these commands normally are invoked, it doesn't prohibit anyone in an edge case to construct​ the cmdline themselves (possibly by modifying one from verbose make gen_xx, if from top-level or build/ is always up to the individual developer and completely besides the point)

I have no idea why I would need to invoke generators separately from the build if not for debugging them. But for debugging this is not really needed because CMake puts generated files first, so one `make` and I see bugs in generators and need to run them manually. And also CMake is nice enough to know (because generators are specified in DEPENDS) when generators need to be rerun, so this covers “generator is in any case invoked only once” case where there is no difference in amount of waiting done in manually vs via cmake cases.

And top-level or build/ is not besides the point, getting actual commands out of the generated Makefile is harder then from top-level which is written by developers. And they tend to be overly verbose: after `make VERBOSE=1` output between highlighted “Generating” and highlighted “Scanning dependencies of target nvim” is unhighlighted, wrapped by the terminal and is literally this:

```
cd /home/zyx/a.a/Proj/c/neovim/build/src/nvim && /usr/bin/clang /home/zyx/a.a/Proj/c/neovim/src/nvim/fileio.c -E -o /home/zyx/a.a/Proj/c/neovim/build/src/nvim/auto/fileio.i -D_GNU_SOURCE -DMIN_LOG_LEVEL=0 -DDO_NOT_DEFINE_EMPTY_ATTRIBUTES -DEXITFREE -I/home/zyx/a.a/Proj/c/neovim/build/config -I/home/zyx/a.a/Proj/c/neovim/src -I/usr/include -I/home/zyx/a.a/Proj/c/neovim/build/src/nvim/auto -I/home/zyx/a.a/Proj/c/neovim/build/include -g -O0 -g -DEXITFREE -Wconversion -O0 -g -DEXITFREE -Wconversion
cd /home/zyx/a.a/Proj/c/neovim/build/src/nvim && /home/zyx/.local-luajit/bin/luajit /home/zyx/a.a/Proj/c/neovim/scripts/gendeclarations.lua /home/zyx/a.a/Proj/c/neovim/src/nvim/fileio.c /home/zyx/a.a/Proj/c/neovim/build/src/nvim/auto/fileio.c.generated.h /home/zyx/a.a/Proj/c/neovim/build/include/fileio.h.generated.h /home/zyx/a.a/Proj/c/neovim/build/src/nvim/auto/fileio.i
cd /home/zyx/a.a/Proj/c/neovim/build && /home/zyx/.local-cmake-v2.8.7/bin/cmake -E cmake_depends "Unix Makefiles" /home/zyx/a.a/Proj/c/neovim /home/zyx/a.a/Proj/c/neovim/src/nvim /home/zyx/a.a/Proj/c/neovim/build /home/zyx/a.a/Proj/c/neovim/build/src/nvim /home/zyx/a.a/Proj/c/neovim/build/src/nvim/CMakeFiles/nvim.dir/DependInfo.cmake --color=
Dependee "/home/zyx/a.a/Proj/c/neovim/src/nvim/fileio.c" is newer than depender "src/nvim/CMakeFiles/nvim.dir/fileio.c.o".
Clearing dependencies in "/home/zyx/a.a/Proj/c/neovim/build/src/nvim/CMakeFiles/nvim.dir/depend.make".
```

Try to find generator invocation here. It is, of course, possible, but if that was written in top-level (not proxied) it would miss `cd` and use relative paths which cuts down text a lot. Though you would have troubles telling CMake invoke it, so such entry point would be for debugging purposes only.